be/src/exec/pipeline/scan_operator.cpp

```diff
@@ -84,6 +87,11 @@ void ScanOperator::set_finishing(RuntimeState* state) {
 }
 
 StatusOr<vectorized::ChunkPtr> ScanOperator::pull_chunk(RuntimeState* state) {
+    if (UNLIKELY(_initial_pickup_morsel)) {
```

Comment:
we have checked `_initial_pickup_morsel` in `ScanOperator::has_output`, why need check again?

Reply:
when _initial_pickup_morsel is true, in pull_chunk, we invoke pickup_morsels, after first pickup_morsels is invoked, _initial_pickup_morsel is set to false. in previous verson, this function is called in ScanOperator::prepare at the first time when runtime filters are not constructed.

Reply:
I think which isn't reasonable, we should only pull chunk when has_output is true.

```
 first pickup_morsels is invoked, _initial_pickup_morsel is set to false
```
why _initial_pickup_morsel need to set false?

Reply:
who triggers first pickup_morsels ?  if none triggers, the has_output return false forever.